src/collective/cover/browser/compose.py

```diff
@@ -0,0 +1,291 @@
+# -*- coding: utf-8 -*-
+from Acquisition import aq_inner
+from collective.cover.tiles.list import IListTile
+from plone import api
+from plone.app.blocks.interfaces import IBlocksTransformEnabled
+from plone.app.uuid.utils import uuidToObject
+from plone.dexterity.events import EditBegunEvent
+from plone.dexterity.utils import createContentInContainer
+from plone.tiles.interfaces import ITileDataManager
+from plone.uuid.interfaces import IUUID
+from plone.uuid.interfaces import IUUIDGenerator
+from Products.Five.browser import BrowserView
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from zExceptions import BadRequest
+from zope.annotation.interfaces import IAnnotations
+from zope.component import getUtility
+from zope.event import notify
+from zope.interface import implementer
+
+import json
+
+
+# TODO: implement EditCancelledEvent and EditFinishedEvent
+# XXX: we need to leave the view after saving or cancelling editing
+@implementer(IBlocksTransformEnabled)
+class Compose(BrowserView):
+
+    index = ViewPageTemplateFile('templates/compose.pt')
+
+    def setup(self):
```

Comment:
@rodfersou in this specific case this is just complicating the code as everything can go inside the `__call__` method